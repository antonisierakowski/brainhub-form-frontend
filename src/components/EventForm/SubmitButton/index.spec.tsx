import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import { SubmitButton } from './index';
import userEvent from '@testing-library/user-event';
import React from 'react';

const cb = jest.fn();

it('should match to the generated snapshot', () => {
  const { asFragment } = render(<SubmitButton onSubmit={cb} isSubmitting={false} />);
  expect(asFragment()).toMatchSnapshot();
});

it('should execute callback once clicked', () => {
  render(
    <SubmitButton onSubmit={cb} isSubmitting={false} />,
  );
  userEvent.click(screen.getByRole('button'));
  expect(cb).toHaveBeenCalledTimes(1);
});
