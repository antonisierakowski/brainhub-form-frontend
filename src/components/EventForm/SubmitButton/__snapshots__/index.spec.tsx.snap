// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match to the generated snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root submitButton MuiButton-contained submitButton MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
